sycl/test-e2e/EnqueueNativeCommand/Graph/cuda_multiple_native_commands.cpp

```diff
@@ -0,0 +1,137 @@
+  int *PtrY = malloc_device<int>(Size, Queue);
+  int *PtrZ = malloc_device<int>(Size, Queue);
+  int *PtrS = malloc_shared<int>(Size, Queue);
+
+  exp_ext::command_graph Graph{Queue};
+
+  Graph.begin_recording(Queue);
+
+  const int ModValue = 42;
+  Queue.submit([&](handler &CGH) {
+    CGH.single_task([=]() {
+      for (size_t i = 0; i < Size; i++) {
+        PtrX[i] = i;
+        PtrY[i] = 0;
+        PtrZ[i] = 0;
+        PtrS[i] = ModValue;
+      }
+    });
+  });
+
+  Queue.submit([&](handler &CGH) {
+    CGH.ext_codeplay_enqueue_native_command([=](interop_handle IH) {
+      if (!IH.ext_codeplay_has_graph()) {
+        assert(false && "Native Handle should have a graph");
+      }
+      CUgraph NativeGraph =
+          IH.ext_codeplay_get_native_graph<backend::ext_oneapi_cuda>();
+
+      CUDA_MEMCPY3D Params;
+      std::memset(&Params, 0, sizeof(CUDA_MEMCPY3D));
+      Params.srcMemoryType = CU_MEMORYTYPE_DEVICE;
+      Params.srcDevice = (CUdeviceptr)PtrX;
+      Params.srcHost = nullptr;
+      Params.dstMemoryType = CU_MEMORYTYPE_DEVICE;
+      Params.dstDevice = (CUdeviceptr)PtrY;
+      Params.dstHost = nullptr;
+      Params.WidthInBytes = Size * sizeof(int);
+      Params.Height = 1;
+      Params.Depth = 1;
+
+      CUgraphNode Node;
+      CUcontext Context = IH.get_native_context<backend::ext_oneapi_cuda>();
+      auto Res = cuGraphAddMemcpyNode(&Node, NativeGraph, nullptr, 0, &Params,
+                                      Context);
+      assert(Res == CUDA_SUCCESS);
+    });
+  });
+
+  Queue.submit([&](handler &CGH) {
+    CGH.host_task([=]() {
+      for (size_t i = 0; i < Size; i++) {
+        PtrS[i] = ModValue * 2;
+      }
+    });
+  });
+
+  Queue.submit([&](handler &CGH) {
+    CGH.parallel_for(range<1>{Size}, [=](id<1> it) { PtrX[it] = PtrS[it]; });
+  });
+
+  Queue.submit([&](handler &CGH) {
+    CGH.ext_codeplay_enqueue_native_command([=](interop_handle IH) {
+      if (!IH.ext_codeplay_has_graph()) {
+        assert(false && "Native Handle should have a graph");
+      }
+      CUgraph NativeGraph =
+          IH.ext_codeplay_get_native_graph<backend::ext_oneapi_cuda>();
+
+      CUDA_MEMCPY3D Params;
+      std::memset(&Params, 0, sizeof(CUDA_MEMCPY3D));
+      Params.srcMemoryType = CU_MEMORYTYPE_DEVICE;
+      Params.srcDevice = (CUdeviceptr)PtrX;
+      Params.srcHost = nullptr;
+      Params.dstMemoryType = CU_MEMORYTYPE_DEVICE;
+      Params.dstDevice = (CUdeviceptr)PtrZ;
+      Params.dstHost = nullptr;
+      Params.WidthInBytes = Size * sizeof(int);
+      Params.Height = 1;
+      Params.Depth = 1;
+
+      CUgraphNode Node;
+      CUcontext Context = IH.get_native_context<backend::ext_oneapi_cuda>();
+      auto Res = cuGraphAddMemcpyNode(&Node, NativeGraph, nullptr, 0, &Params,
+                                      Context);
+      assert(Res == CUDA_SUCCESS);
+    });
+  });
+
+  Graph.end_recording();
+
+  auto ExecGraph = Graph.finalize();
+  Queue.ext_oneapi_graph(ExecGraph).wait();
```

Comment:
Do we need to wait here since Queue is in-order?

Reply:
nope, removed